lib/util/path.js

```diff
@@ -201,3 +200,16 @@ const cachedJoin = (rootPath, request) => {
 	return cacheEntry;
 };
 exports.cachedJoin = cachedJoin;
+
+/**
+ * @param {string} rawPath node.js path
+ * @returns {string} normalized for windows, same for posix paths
+ */
+const transformPathToPosix = rawPath => {
+	if (!rawPath) return rawPath;
+
+	if (path.sep === "/")
+		return rawPath; // Don't even transform if path.sep is posix
+	else return rawPath.split("\\").join("/");
+};
```

Comment:
I agree about the performance part. About the behavior: it's already different and based on os.  

It's possible with this PR to test how different behavior between Unix and Windows with the following instruction:
- revert changes from library itself, keep the changes for tests
- mock `transferPathToPosix` function in `test/util/path-separator.js` because it exists in tests only for assertion and handles assertion for changes exclusively in current pull request (it removes the conversion from `\\` to `/` provided by changes in library which we reverted)